src/library_webgpu.js

```diff
@@ -1495,9 +1497,10 @@ var LibraryWebGPU = {
     {{{ gpu.makeCheckDescriptor('descriptorFromHTMLCanvasId') }}}
     var idPtr = {{{ makeGetValue('descriptorFromHTMLCanvasId', C_STRUCTS.WGPUSurfaceDescriptorFromHTMLCanvasId.id, '*') }}};
     {{{ gpu.makeCheck('idPtr') }}}
-    var id = UTF8ToString(idPtr);
-    var canvas = document.getElementById(id);
+    var canvas = __findCanvasEventTarget(idPtr);
```

Comment:
Old code was relying on old deprecated canvas lookup rules, this function works for both old and new.

Reply:
I don't think getElementById is considered to be deprecated. When choosing how this would would I specifically picked this as it was the most straightforward way to get at most one result. I guess querySelector is fine since it also guarantees that. But if we're going to change the behavior here we should change the naming in the header as well ("FromHTMLCanvasId") which I think we should do at the same time (and first upstream: https://github.com/webgpu-native/webgpu-headers).

Reply:
Oh sorry, I did not mean that `.getElementById` itself would be deprecated, but that in Emscripten we have migrated from using IDs to using query selectors in other APIs to choose canvas targets, as that is more flexible. (one can target by DOM element name, or by ID, or by CSS class, or by attribute, etc.) Would be nice to be consistent here?

Reply:
Thanks for the context, filed https://github.com/webgpu-native/webgpu-headers/issues/54